src/main/java/org/testng/internal/ITestInvoker.java

```diff
@@ -30,4 +31,15 @@ FailureContext retryFailed(
       ITestContext testContext);
 
   void runTestResultListener(ITestResult tr);
+
+  ITestResult registerSkippedTestResult(ITestNGMethod testMethod, long start, Throwable throwable);
+
+  void invokeListenersForSkippedTestResult(ITestResult r, IInvokedMethod invokedMethod);
+
+  ITestResultNotifier getNotifier();
+
+  default IMethodRunner getRunner() {
```

Comment:
Why do you use a default method here, instead of a classical interface/implementation logic?

Reply:
> Why do you use a default method here,

I chose to use a `default` method over mere convenience via its default constructor and nothing else. The other implementations viz., `TestInvoker` and `ConfigInvoker` need parameters and so I couldn't resort to a `default` implementation.

> instead of a classical interface/implementation logic?

This is still interface driven (`MethodRunner` implements `IMethodRunner` interface).

```java
class MethodRunner implements IMethodRunner {
```

Reply:
Yes. You are right. I didn't choose the abstract route 

1. `AbstractTestInvoker` houses all required logic
2. `SequentialTestInvoker extends AbstractTestInvoker`
3. `ParallelTestInvoker extends AbstractTestInvoker`

because this would basically need me to expose these classes outside of the `internal` package. If it's the interface route, then we can let downstream users tweak in if they needed.